File: backend/ai-service/src/ai_service/api/health.py
import time
from datetime import datetime, timezone

from fastapi import APIRouter

router = APIRouter()

_started_at = time.monotonic()


@router.get("/health")
def health() -> dict[str, object]:
    return {
        "status": "ok",
        "service": "ai-service",
        "timestamp": datetime.now(timezone.utc).isoformat(),
        "uptime": time.monotonic() - _started_at,
    }

File: backend/realtime-service/internal/api/health.go
package api

import (
	"encoding/json"
	"net/http"
	"time"
)

var startedAt = time.Now()

type healthResponse struct {
	Status    string  `json:"status"`
	Service   string  `json:"service"`
	Timestamp string  `json:"timestamp"`
	Uptime    float64 `json:"uptime"`
}

func HealthHandler(w http.ResponseWriter, _ *http.Request) {
	resp := healthResponse{
		Status:    "ok",
		Service:   "realtime-service",
		Timestamp: time.Now().UTC().Format(time.RFC3339),
		Uptime:    time.Since(startedAt).Seconds(),
	}
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	_ = json.NewEncoder(w).Encode(resp)
}
